index.js

```diff
@@ -42,6 +42,8 @@ const getStreamContents = async (stream, {convertChunk, getContents}, {maxBuffer
 	}
 };
 
+const isAsyncIterable = stream => typeof stream === 'object' && stream !== null && typeof stream[Symbol.asyncIterator] === 'function';
```

Comment:
We check whether the argument is async iterable (instead of being a stream) so that this works as well with web streams. Also, the logic currently just uses `for await`.